Develop/parser

mpdesantis opened the pull request:
Functional Parser and ParserTest classes.
Main updated to generate input events from input file through parser.
Minor doc and grammar edits.

Thanks.

VictoriaMalouf committed:
project structure

mpdesantis committed:
md: Initialize Parser class.

VictoriaMalouf committed:
method headers

mpdesantis committed:
md: Update doc.

VictoriaMalouf committed:
subsystem references

orselane committed:
Updating readme with scope assumptions

orselane committed:
Updating readme with my information

- Student ID
- Task for iteration 0

VictoriaMalouf committed:
systemevent class

ArthurAtangana committed:
add tasks table to README.md

mpdesantis committed:
Move Parser code to FloorSubsytem class.

mpdesantis committed:
Initialize input file.

mpdesantis committed:
Implement Parser's time string to long conversion.

mpdesantis committed:
Initialize JUnit test class for Parser.

mpdesantis committed:
Update Passenger arrivalTime to a long.

mpdesantis committed:
Implement Parser's stringToPassenger method and test.

mpdesantis committed:
Implement Parser's fileToPassengers method and test.

mpdesantis committed:
Update input file.

mpdesantis committed:
Code clean.

mpdesantis committed:
Signature change for file parse method.

mpdesantis committed:
Update calls with new arguments.

mpdesantis committed:
Code clean.

mpdesantis committed:
Code clean.

mpdesantis committed:
Code clean.

mpdesantis committed:
Move forgotten instance variable to local variable.

mpdesantis committed:
Fix conflicts (hopefully).

mpdesantis committed:
Fix more conflicts (hopefully).

mpdesantis committed:
Fix conflicts.

mpdesantis committed:
Replace Passenger record with FloorInputEvent record.

mpdesantis committed:
Privatize methods. Remove math.

mpdesantis committed:
Update test class for modified methods.

mpdesantis committed:
Move ParserTest to unit test module.

mpdesantis committed:
Incorporate parser into Main, start dispatcher with parsed input objects.

mpdesantis committed:
Minor grammar edits to validate my English degrees.

mpdesantis committed:
Minimal doc headers added.

braedenkloke reviewed (commented):
Looks great! Test and Main function working as expected.

Before I approve. I'm going to refactor the ParserTest class to the `test` folder.

braedenkloke committed:
refactor(ParserTest): moved class ParserTest to test folder

braedenkloke committed:
fix(test): removed deprecated tests

braedenkloke committed:
docs(parser): add todo item for variable file path

Parser should be accept files from different folders.
Currently filepath is hardcoded as /src/FloorSubsystem.
Should be updated in future to accept input files from different folders.

braedenkloke reviewed (approved):
Looks good to me.

Only note I added to Parser was a TODO item specifying the file path to be variable.
Currently file path is hardcoded as src/FloorSubystem

orselane commented:
The variable filepath is a known issue, part of the effort of #11
